torchrl/data/tensor_specs.py

```diff
@@ -469,7 +469,7 @@ def encode(
         if (val >= space.n).any():
             raise AssertionError("Value must be less than action space.")
 
-        val = torch.nn.functional.one_hot(val, space.n).to(torch.long)
+        val = torch.nn.functional.one_hot(val.long(), space.n).to(torch.long)
```

Comment:
When running `test_mult_discrete_action_spec_reconstruct`, this explicit cast was required for `one_hot` to accept its input (line 993 - `actions_tensors = [action_spec.encode(a) for a in actions_numpy]`)

Reply:
do we need the `to(long)` in the end then?

Reply:
No indeed. Removed